File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/MessageDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.extern.jackson.Jacksonized;

import java.util.UUID;

@Builder
@Jacksonized
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MessageDTO {
    UUID id;
    String from;
    String to;
    String text;
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/MessageDTO.kt
package com.szlazakm.safechat.webclient.dtos

import java.time.Instant

data class MessageDTO(
    val from: String,
    val to: String,
    val text: String,

    val phoneNumber: String,
    val nonceTimestamp: Long,
    val nonce: ByteArray,
    val authMessageSignature: ByteArray
)
